### ISSUE TITLE
Unable to build iOS framework for OpenCV 3.4.5

### ISSUE THREAD

**Issue by solanki21** [opencv/opencv#22487]
##### System information (version)
- OpenCV => 3.4.5
- Operating System / Platform => Mac/iOS
- Compiler => Xcode

##### Detailed description
I checked out OpenCV (and contrib) version 3.4.5.
Trying to run python `opencv/platforms/ios/build_framework.py ios`

But getting following error:
```
Using IPHONEOS_DEPLOYMENT_TARGET=8.0
Using iPhoneOS ARCHS=['armv7', 'armv7s', 'arm64']
Using iPhoneSimulator ARCHS=['i386', 'x86_64']
============================================================
ERROR: cannot use a string pattern on a bytes-like object
============================================================
Traceback (most recent call last):
  File "opencv/platforms/ios/build_framework.py", line 117, in build
    self._build(outdir)
  File "opencv/platforms/ios/build_framework.py", line 85, in _build
    xcode_ver = getXCodeMajor()
  File "opencv/platforms/ios/build_framework.py", line 45, in getXCodeMajor
    m = re.match(r'Xcode\s+(\d+)\..*', ret, flags=re.IGNORECASE)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/re.py", line 191, in match
    return _compile(pattern, flags).match(string)
TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by alalek** [comment]
> 3.4.5

You are trying to use outdated OpenCV version.
We don't amend already released code, so you need to upgrade (or use Xcode and other build environment from 2018 too).
This problem is already fixed in the latest release: #22054

Build / usage questions should go to Users OpenCV Q/A forum: https://forum.opencv.org/

**Comment by solanki21** [comment]
Ok thanks.
